extension/test/unit/config/config.spec.js

```diff
@@ -314,4 +314,124 @@ describe('::config::', () => {
       })
     }
   )
+
+  it(
+    'when removeSensitiveData is set as boolean false in config.js, ' +
+      'it should load as false value in module config',
+    () => {
+      process.env.ADYEN_INTEGRATION_CONFIG = JSON.stringify({
+        commercetools: {
+          ctpProjectKey1: {
+            clientId: 'clientId',
+            clientSecret: 'clientSecret',
+            apiUrl: 'host',
+            authUrl: 'authUrl',
+            authentication: {
+              scheme: 'basic',
+              username: 'username',
+              password: 'password',
+            },
+          },
+        },
+        adyen: {
+          adyenMerchantAccount1: {
+            apiBaseUrl: 'apiBaseUrl',
+            apiKey: 'apiKey',
+            clientKey: 'clientKey',
+            legacyApiBaseUrl: 'legacyApiBaseUrl',
+          },
+        },
+        logLevel: 'DEBUG',
+        removeSensitiveData: false,
+      })
+      const config = requireUncached('../../../src/config/config')
+      expect(config.getModuleConfig()?.removeSensitiveData).to.eql(false)
```

Comment:
```suggestion
      expect(config.getModuleConfig().removeSensitiveData).to.eql(false)
```

Remove it everywhere.